### PR TITLE
Changed timestamp of item insert from MySql server time to local Java…

### PR DESCRIPTION
**Pull request by herby3**
see #4495

### CODE REVIEW

File: bundles/persistence/org.openhab.persistence.mysql/java/org/openhab/persistence/mysql/internal/MysqlPersistenceService.java [hunk 1]
```diff
@@ -421,15 +421,20 @@ public void store(Item item, String alias) {
         String sqlCmd = null;
         PreparedStatement statement = null;
         try {
+            // Get current timestamp
+            long timeNow = Calendar.getInstance().getTimeInMillis();
+    	    java.sql.Timestamp timestamp = new java.sql.Timestamp(timeNow);
```
Comment (watou) [inline bundles/persistence/org.openhab.persistence.mysql/java/org/openhab/persistence/mysql/internal/MysqlPersistenceService.java]: Indentation (consider using the Eclipse IDE to auto-format code).  Also, please import java.sql.Timestamp and use here without package qualifier.

**Comment by watou** [comment]
Thanks for this.  

What are the potential effects on existing users of MySQL persistence?  If MySQL server and openHAB server times are very different, there will be a time leap in persisted item states in rows inserted after this change is running, which may be unwanted.  Perhaps a compatibility flag for openhab.cfg / mysql.cfg, and the code would use the time source the user specified?

**Comment by 9037568** [comment]
I like the idea of making this user configurable.  Beneficial for those who
want to continue to use the server time...

**Comment by watou** [comment]
Maybe a `#mysql:localtime=true` setting which defaults to `false` if not specified, so that an upgrade does not surprisingly change how times are stored?

**Comment by herby3** [comment]
I added flag 'localtime' to configuration and it defaults to 'false'.

**Comment by watou** [comment]
Excellent, thanks @herby3.  Could you also update [mysql.cfg](https://github.com/openhab/openhab/blob/master/features/openhab-addons-external/src/main/resources/conf/mysql.cfg) and also [the wiki page](https://github.com/openhab/openhab/wiki/MySQL-Persistence) (noting in the wiki page that this is a 1.9 enhancement)?  With that, and your testing that it works as intended, we should be ready to merge this PR.

File: features/openhab-addons-external/src/main/resources/conf/mysql.cfg [hunk 2]
```diff
@@ -11,4 +11,9 @@
...
 
 # the connection timeout (in seconds)
-#waitTimeout=
\ No newline at end of file
+#waitTimeout=
+
+# Use MySQL Server time to store item values (=false) or use openHAB Server time (=true).
+# For new installations, its recommend to set "mysql:localtime=true".
+# (optional, defaults to false)
+#mysql:localtime=true
```
Comment (watou) [inline features/openhab-addons-external/src/main/resources/conf/mysql.cfg]: The `mysql:` part is not used in `mysql.cfg`.

File: features/openhab-addons-external/src/main/resources/conf/mysql.cfg [hunk 3]
```diff
@@ -11,4 +11,9 @@
 #reconnectCnt=
 
 # the connection timeout (in seconds)
-#waitTimeout=
\ No newline at end of file
+#waitTimeout=
+
+# Use MySQL Server time to store item values (=false) or use openHAB Server time (=true).
+# For new installations, its recommend to set "mysql:localtime=true".
```
Comment (watou) [inline features/openhab-addons-external/src/main/resources/conf/mysql.cfg]: also remove `mysql:` here as well

**Comment by herby3** [comment]
Sry, done!

**Comment by watou** [comment]
Thanks @herby3!

**Pull request merged**